File: Data Structures & Algorithms/max-water-container/submission-1.cpp
class Solution {
public:
    int maxArea(vector<int>& nums) {
     int a=0;
     int i=0,j=nums.size()-1;
     while(i<j){
        int c=min(nums[i],nums[j])*(j-i);
        if(nums[i]>nums[j])j--;
        else i++;
        a=max(c,a);
     }  
     return a; 
    }
};

File: Data Structures & Algorithms/eating-bananas/submission-1.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        large=max(piles)
        i=1
        n=len(piles)
        # sum=0
        ans=0
        if n==h :
            return large
        j=large    
        while i<=j:
            m=i+(j-i)//2
            sum=0
            for a in piles:
                sum+=(a+m-1)//m
            if sum>h :
                 i=m+1
            else :
                j=m-1 
                ans=m
        return ans          
